ksql-engine/src/main/java/io/confluent/ksql/util/ExpressionTypeManager.java

```diff
@@ -265,11 +271,28 @@ protected Expression visitFunctionCall(
     return null;
   }
 
-  private static Schema resolveArithmeticType(final Schema leftSchema,
-      final Schema rightSchema) {
+  private static Schema resolveArithmeticType(final Schema leftSchema, final Schema rightSchema) {
     return SchemaUtil.resolveArithmeticType(leftSchema.type(), rightSchema.type());
   }
 
+  private static void isValidComparison(
```

Comment:
As I said previously, this whole function could be part of `ComparisonUtil` and suitable tests added to its test class. Much easier to unit test the util class.

Reply:
Had to move `ComparisonUtil` from `ksql-common` to `ksql-engine`. Now `isValidComparison` is in 'ComparisonUtil` as you suggested.